lib/perl/Genome/Config/AnalysisProject/Command/DiskUsage.pm

```diff
@@ -0,0 +1,101 @@
+package Genome::Config::AnalysisProject::Command::DiskUsage;
+
+use strict;
+use warnings;
+
+use feature qw(say);
+
+use Genome;
+
+class Genome::Config::AnalysisProject::Command::DiskUsage {
+    is => 'Command::V2',
+    has => [
+        analysis_projects => {
+            is                  => 'Genome::Config::AnalysisProject',
+            doc                 => 'the analysis projects to report disk usage',
+            is_many             => 1,
+            shell_args_position => 1,
+        },
+    ],
+    doc => 'report on the disk usage of analysis projects',
+};
+
+sub help_detail {
+    return <<EOHELP
+This command will take several minutes to run.
+
+The output is a tab-delimited disk usage summary for each model type and configuration item per analysis project.
+The colums of the output file are:
+analysis project ID, configuration item ID, subclass name, model count, build count, total KB, total base pair, bytes per base
+
+EOHELP
+}
+
+sub execute {
+    my $self = shift;
+
+    my @data;
+    for my $anp ($self->analysis_projects) {
+        for my $config_item ($anp->config_items) {
+            my %model_type_disk_usage;
+            my $model_count = 0;
+            my $build_count = 0;
+            for my $model ($anp->models) {
+                my $subclass_name = $model->subclass_name;
+                $model_type_disk_usage{$subclass_name}{'model_count'}++;
+                for my $build ($model->builds) {
+                    $model_type_disk_usage{$subclass_name}{'build_count'}++;
+                    my @allocations = $build->disk_usage_allocations;
+                    for my $allocation (@allocations) {
+                        $model_type_disk_usage{$subclass_name}{'allocations'}{$allocation->id} = $allocation->kilobytes_used;
```

Comment:
One corner case to be aware of: `kilobytes_used` will be `0` for any allocations that have never reallocated.  (For example, new allocations from running builds.)